### ISSUE TITLE
Carousel options do not respond to state change

### ISSUE THREAD

**Issue by thearthurdev** [serenader2014/flutter_carousel_slider#142]
I have a boolean variable that updates when the device switches between landscape and portrait mode, then I use ternary operators for different values for the carousel options to get a more responsive UI. 

However, the carousel does not seem to respond to the changes in the variable that tracks device orientation. calling `setState((){})` does not work. The only way I can get the carousel to reflect the change in state is to completely rebuild the widget. 

This is the code for something similar to what I'm trying to achieve. In the code below I use the FAB to swap out and back in the carousel before it reflects the desired changes.

I have also screen recorded the undesired behaviour [here](https://drive.google.com/open?id=1n3hVHtGmPhixCyT8jg24QddDQ-91oWl0) 

```
import 'package:carousel_slider/carousel_slider.dart';
import 'package:flutter/material.dart';

class TestPage extends StatefulWidget {
  static const String id = '/TestPage';

  @override
  _TestPageState createState() => _TestPageState();
}

class _TestPageState extends State<TestPage> {
  bool showCarousel;

  @override
  void initState() {
    showCarousel = true;
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    bool isLandscape =
        MediaQuery.of(context).orientation == Orientation.landscape;

    return Scaffold(
      appBar: AppBar(
        title: Text('Carousel Slider'),
        centerTitle: true,
      ),
      floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
      floatingActionButton: FloatingActionButton.extended(
        onPressed: () {
          setState(() {
            showCarousel = !showCarousel;
          });
        },
        label: Text('Swap'),
        icon: Icon(Icons.refresh),
      ),
      body: Container(
        child: Center(
          child: showCarousel
              ? CarouselSlider.builder(
                  itemCount: 9,
                  itemBuilder: (context, i) {
                    return Container(
                      width: 300.0,
                      height: 400.0,
                      padding: EdgeInsets.all(16.0),
                      color: Colors.primaries.reversed.toList()[i],
                    );
                  },
                  options: CarouselOptions(
                    height: 500.0,
                    autoPlay: true,
                    autoPlayCurve: Curves.easeOut,
                    viewportFraction: isLandscape ? 0.3 : 1.0,
                    enlargeCenterPage: isLandscape,
                  ),
                )
              : Text('Swapping...'),
        ),
      ),
    );
  }
}

```

**Comment by serenader2014** [comment]
Hi @ArthurDevv , thank you for reporting this issue. This indeed is a problem :(, I'll take a look at it. In the meanwhile, if you have some thought on this, or if you know how to fix this, PR is always welcome :)

**Comment by ArthurDevv** [comment]
I can't think of a solution off the top of my head, I'm still relatively new to flutter. I'll fork the repo and see if I can find a fix. Wish me luck!

**Comment by thearthurdev** [comment]
@serenader2014 I have (hopefully) fixed the issue and submitted a PR. I can confirm it is working on my end. 

Please review and let me know so I may close the issue.

**Comment by rgp10** [comment]
Hi, was this fixed in the end? I'm having a similar issue

**Comment by thearthurdev** [comment]
I have fixed the issue in my fork of the package. I'm awaiting my PR to be merged. 

See the comment below for the link to the fork and and how to use it in your project 
https://github.com/serenader2014/flutter_carousel_slider/issues/143#issuecomment-629124045

**Comment by rgp10** [comment]
Awesome, Cheers!

**Comment by erabti** [comment]
This should be fixed asap

**Comment by DFelten** [comment]
Same issue here. I hope this will be fixed soon.

**Comment by serenader2014** [comment]
Sorry for taking so long to merge the MR, I've been so busy these months. The MR is merged, and published in v2.3.0, please have a try.

**Comment by thearthurdev** [comment]
Thanks for merging the PR, better late than never right?
